### DISCUSSION TITLE
[Implemented] Ability to add effect to armour (and/or tools)

### DISCUSSION THREAD

**Discussion by mrfloris** [Zrips/CMI#1370] [Enhancement]
Hey there,

It would be quite great if the /cmi effect could get extended to "hand", so what we're holding, it might be able to put the effect as an enchant on the armour bit, or weapon, or tool?

This way, together with a permission like - cmi.effect.armor.use or whatever, we can give people custom armor, tools, and what not that when placed on them, or when used, could give an effect. Such as night vision.

This would allow us to create some fun suits for mining that could give a 60 seconds night vision effect.

Optionally, 

-t:50 (50 seconds that it is applied?) and -c:300 (300 seconds cooldown before it could work again)

**Comment by mrfloris** [comment]
spoiler: #soon

**Comment by mrfloris** [comment]
This has now been implemented in 9.2.4.x

**Comment by CloudeLecaw** [comment]
you cab add an effect to an armor?
how?
an example would be great :)

edit i found the patchlog from 9.2.4.x

```Added /cmi armoreffect command which can attach potion effect to any item. Keep in mind that effect only gets applied when item is in armor slot (offhand excluded). This will apply defined potion effect for 15 seconds and keep reapplying it while you keep wearing this armor. Example of usage: /cmi armoreffect jump, /cmi armoreffect jump:2, /cmi armoreffect jump:2:v Providing number will define effect level and v will define if you want to have potion particles around player when this effect is active, by default there won't be any particles shown and no icon on top right corner to indicate potion effect being active. You can use /cmi armoreffect which will show you basic chat editor in same way it is for attachedcommands where you can have better control over attached potion effects or modify existing ones. To apply more than one effect use ;; like /cmi armoreffect jump:2;;regeneration:v ```

so i think no example is needed anymore

**Reply by CloudeLecaw** [reply]
so one more question..was the offhand added allready ?
